ims/einstein/iscsi/mock.py

```diff
@@ -0,0 +1,124 @@
+import ims.exception.iscsi_exceptions as iscsi_exceptions
+from ims.interfaces.iscsi import ISCSI
+
+
+class MockISCSI(ISCSI):
+    """
+    This is mock class that implements most of the abstract methods for ISCSI
+    class. It returns mostly true or success or some exception. This spoofs the
+    requests coming in. This should not have any dependency on file system or
+    database or any other component as we are writing unit tests for iSCSI
+    server alone. For example, there may be an iSCSI server that implements
+    adding target by writing to database instead of files. So, the idea here
+    is to touch only services that are core to iSCSI. Other exceptions like
+    touching files(for updating configurations) etc, should be handled in
+    integration tests or unit tests for that particular iSCSI server. This
+    will have helper methods for setting target_lists and server status.
+    """
+    def __init__(self):
+        # Initial target list
+        self.target_list = []
+        # Initial state of iSCSI server
+        self.server_status = ""
+
+    def set_target_list(self, target_list):
+        """
+        Setter method for target list to be used by test cases.
+        Args:
+            target_list: List provided by user
+        Returns:
+            None, will update the list provided.
+        """
+        self.target_list += target_list
+
+    def set_server_status(self, status):
+        """
+        Setter method for iSCSI server to be used by test cases.
+        Args:
+            status: String which has to be either Running or Dead
+        Returns:
+            Either error state or updates the status of mock iSCSI server
+        """
+        # TODO: make states constants across all iSCSI implementations.
+        valid_states = ["Running", "Dead"]
+        error = "iSCSI server in unknown state"
+        if status in valid_states:
+            self.server_status = status
+        else:
+            return error
```

Comment:
I think its better to have a server unknown state exception. Whatever I mentioned in TODO for state change can be used for it.